sdc/hiframes/boxing.py

```diff
@@ -100,87 +79,46 @@ def typeof_pd_str_series(val, c):
         _infer_series_dtype(val), index=index_type, is_named=is_named)
 
 
-@intrinsic
-def my_intrinsic(typingctx, val):
-    col_list_type = types.List(string_type)
-
-    # An intrinsic that returns `list`
-    def impl(context, builder, typ, args):
-        # return context.build_list(builder, col_list_type, [arg for arg in args])
-        return context.build_list(builder, col_list_type, args[0])
-
-    sig = signature(col_list_type, val)
-
-    return sig, impl
-
-
 @unbox(DataFrameType)
 def unbox_dataframe(typ, val, c):
     """unbox dataframe to an empty DataFrame struct
     columns will be extracted later if necessary.
     """
     n_cols = len(typ.columns)
-    data_typs = typ.data
-    column_names = typ.columns
-
-    # Store unique types of columns ex. {'int64': (0, [0, 2]), 'float64': (1, [1])}
-    data_typs_map = {}
-    types_order = []
-    type_id = 0
-    for col_id, col_typ in enumerate(data_typs):
-        if col_typ not in data_typs_map:
-            data_typs_map[col_typ] = (type_id, [col_id])
-            # The first column in each type always has 0 index
-            types_order.append(col_typ)
-            type_id += 1
-        else:
-            type_idx, col_indices = data_typs_map[col_typ]
-            col_indices.append(col_id)
-
     column_strs = [numba.cpython.unicode.make_string_from_constant(
         c.context, c.builder, string_type, a) for a in typ.columns]
-
     # create dataframe struct and store values
     dataframe = cgutils.create_struct_proxy(typ)(c.context, c.builder)
 
-    column_tup = c.context.make_tuple(
-        c.builder, types.UniTuple(string_type, n_cols), column_strs)
-
-    def wrapper(args):
-        return my_intrinsic(args)
-
-    args = [column_tup]
     col_list_type = types.List(string_type)
-    resty = col_list_type
-    uni_tup = types.UniTuple(string_type, n_cols)
-    sig = typing.signature(resty, uni_tup)
+    _, inst = listobj.ListInstance.allocate_ex(c.context, c.builder, col_list_type, n_cols)
```

Comment:
Maybe handle allocating of the list something like that:
```python
errorptr = cgutils.alloca_once_value(c.builder, cgutils.false_bit)
ok, inst = listobj.ListInstance.allocate_ex(c.context, c.builder, col_list_type, n_cols)
with c.builder.if_else(ok, likely=True) as (if_ok, if_not_ok):
    with if_ok:
        # code where inst is used
    with if_not_ok:
        c.builder.store(cgutils.true_bit, errorptr)

# If an error occurred, drop the whole native list
with c.builder.if_then(c.builder.load(errorptr)):
    c.context.nrt.decref(c.builder, col_list_type, inst.value)
```

Reply:
done